test_tf2/CMakeLists.txt

```diff
@@ -36,92 +35,88 @@ ament_find_gtest()
 
 ament_add_gtest(buffer_core_test test/buffer_core_test.cpp)
 if(TARGET buffer_core_test)
-  ament_target_dependencies(buffer_core_test
-    builtin_interfaces
-    geometry_msgs
-    rclcpp
-    tf2
-    tf2_geometry_msgs
-    tf2_ros
-  )
+  target_link_libraries(buffer_core_test
+    ${builtin_interfaces_TARGETS}
+    ${geometry_msgs_TARGETS}
+    rclcpp::rclcpp
+    tf2::tf2
+    ${tf2_geometry_msgs_TARGETS}
+    tf2_ros::tf2_ros)
 endif()
 
 ament_add_gtest(test_message_filter test/test_message_filter.cpp)
 if(TARGET test_message_filter)
-  ament_target_dependencies(test_message_filter
-    builtin_interfaces
-    geometry_msgs
-    rclcpp
-    tf2
-    tf2_ros
-  )
+  target_link_libraries(test_message_filter
+    ${builtin_interfaces_TARGETS}
+    ${geometry_msgs_TARGETS}
+    rclcpp::rclcpp
+    tf2::tf2
+    tf2_ros::tf2_ros)
 endif()
 
 ament_add_gtest(test_convert test/test_convert.cpp)
 if(TARGET test_convert)
-  ament_target_dependencies(test_convert
-    Eigen3
-    eigen3_cmake_module
-    tf2
-    tf2_bullet
-    tf2_eigen
-    tf2_geometry_msgs
-    tf2_kdl
-  )
+  target_link_libraries(test_convert
+    ${geometry_msgs_TARGETS}
+    tf2::tf2
+    tf2_bullet::tf2_bullet
+    tf2_eigen::tf2_eigen
+    ${tf2_geometry_msgs_TARGETS}
+    tf2_kdl::tf2_kdl)
+    if(TARGET Eigen3::Eigen)
+      # TODO(sloretz) require target to exist when https://github.com/ros2/choco-packages/issues/19 is addressed
+      target_link_libraries(test_convert Eigen3::Eigen)
+    else()
+      target_include_directories(test_convert ${Eigen3_INCLUDE_DIRS})
```

Comment:
Windows CI is complaining; I think you need `PRIVATE` here.

Reply:
Added `PRIVATE` in 06a70c708efcb48283985ef02d3aa5fe36a542ef